GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRCameraSceneObject.java

```diff
@@ -81,29 +91,166 @@ public GVRCameraSceneObject(GVRContext gvrContext, GVRMesh mesh,
-     *            calls {@link Camera#setPreviewTexture(SurfaceTexture)} so you
-     *            should be sure to call it before you call
-     *            {@link Camera#startPreview()}.
+     *
+     * @param gvrContext current {@link GVRContext}
+     * @param width      the scene rectangle's width
+     * @param height     the rectangle's height
+     * @param camera     an Android {@link Camera}. <em>Note</em>: this constructor
+     *                   calls {@link Camera#setPreviewTexture(SurfaceTexture)} so you
+     *                   should be sure to call it before you call
+     *                   {@link Camera#startPreview()}.
+     * @deprecated This call does not ensure the activity lifecycle is correctly
+     * handled by the {@link GVRCameraSceneObject}. Use
+     * {@link #GVRCameraSceneObject(GVRContext, float, float)} instead.
      */
     public GVRCameraSceneObject(GVRContext gvrContext, float width,
-            float height, Camera camera) {
+                                float height, Camera camera) {
         this(gvrContext, gvrContext.createQuad(width, height), camera);
     }
 
     /**
+     * Create a 2D, rectangular {@linkplain GVRSceneObject scene object} that
+     * shows live video from one of the device's cameras.
+     *
+     * @param gvrContext current {@link GVRContext}
+     * @param width      the scene rectangle's width
+     * @param height     the rectangle's height
+     *
+     * @throws GVRCameraAccessException this exception is returned when the camera cannot be opened.
+     */
+    public GVRCameraSceneObject(GVRContext gvrContext, float width,
+                                float height) throws GVRCameraAccessException {
+        this(gvrContext, gvrContext.createQuad(width, height));
+    }
+
+    private boolean openCamera() {
+        if (camera != null) {
+            //already open
+            return true;
+        }
+
+        if (!checkCameraHardware(gvrContext.getActivity())) {
+            android.util.Log.d(TAG, "Camera hardware not available.");
+            return false;
+        }
+        try {
+            camera = Camera.open();
+
+            if (camera == null) {
+                android.util.Log.d(TAG, "Camera not available or is in use");
+                return false;
+            }
+            camera.startPreview();
+            camera.setPreviewTexture(mSurfaceTexture);
+            isCameraOpen = true;
+        } catch (Exception exception) {
+            android.util.Log.d(TAG, "Camera not available or is in use");
+            return false;
+        }
+
+        return true;
+    }
+
+    private void closeCamera() {
+        if (camera == null) {
+            //nothing to do
+            return;
+        }
+
+        camera.stopPreview();
+        camera.release();
+        camera = null;
+        isCameraOpen = false;
+    }
+
+    private IActivityEvents activityEvents = new IActivityEvents() {
```

Comment:
Fyi, there is empty implementation in GVREventListeners.ActivityEvents that can be used to eliminate the noop methods.